service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/shiro/AuthenticationServiceShiroImpl.java

```diff
@@ -477,4 +489,34 @@ private String generateJwt(KapuaEid scopeId, KapuaEid userId, Date now, long ttl
         return jwt;
     }
 
+    private Boolean checkIfCredentialHasJustBeenLocked(AuthenticationToken authenticationToken) throws KapuaException {
+        String principal = (String) authenticationToken.getPrincipal();
+        User user = KapuaSecurityUtils.doPrivileged(() -> userService.findByName(principal));
+        Credential credential = null;
+        if (user != null) {
+            credential = KapuaSecurityUtils.doPrivileged(() -> {
+                CredentialListResult credentialList = credentialService.findByUserId(user.getScopeId(), user.getId());
+
+                if (credentialList != null && !credentialList.isEmpty()) {
```

Comment:
`CredentialService.findByUserId()` will never return `null` as result . So this line can be removed.